tests/tt_eager/profiling/profile_host_overhead.py

```diff
@@ -0,0 +1,294 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import pytest
+import torch
+import tt_lib
+import time
+import statistics
+from loguru import logger
+from models.utility_functions import torch2tt_tensor
+from tests.tt_eager.profiling import ops_for_profiling
+from tracy import signpost
+
+
+test_sweep_args = [
+    # (
+    #     (1, 2, 1024, 1024),
+    #     tt_lib.tensor.DataType.BFLOAT16,
+    #     tt_lib.tensor.Layout.TILE,
+    #     tt_lib.tensor.MemoryConfig(tt_lib.tensor.TensorMemoryLayout.INTERLEAVED, tt_lib.tensor.BufferType.DRAM),
+    #     tt_lib.tensor.MemoryConfig(tt_lib.tensor.TensorMemoryLayout.INTERLEAVED, tt_lib.tensor.BufferType.DRAM),
+    # ),
+    (
+        (1, 4, 1024, 1024),
+        tt_lib.tensor.DataType.BFLOAT16,
+        tt_lib.tensor.Layout.TILE,
+        tt_lib.tensor.MemoryConfig(tt_lib.tensor.TensorMemoryLayout.INTERLEAVED, tt_lib.tensor.BufferType.DRAM),
+        tt_lib.tensor.MemoryConfig(tt_lib.tensor.TensorMemoryLayout.INTERLEAVED, tt_lib.tensor.BufferType.DRAM),
+    ),
+]
+
+all_num_call_to_stack = [1, 3]  # For 10 and more test  execution spills to dispatch
+num_repeats = 5
+
+
+def measure_host_overhead(op_func, op_name, device, num_call_to_stack, is_warmup):
+    if not is_warmup:
+        signpost(header=f"start {op_name}")
+
+    start_time = time.time()
+    for _ in range(num_call_to_stack):
+        op_func()
+
+    tt_lib.device.Synchronize(device)
+
+    duration = 1000 * (time.time() - start_time)
+    total_op_time = duration / num_call_to_stack
+    logger.info(f"{num_call_to_stack} calls and Synchronize after {duration:.2f}ms ({total_op_time:.2f}ms per call)")
+
+    start_time = time.time()
+    for _ in range(num_call_to_stack):
+        op_func()
+
+    duration = 1000 * (time.time() - start_time)
+    overhead_ms = duration / num_call_to_stack
+    logger.info(f"{num_call_to_stack} calls without Synchronize {duration:.2f}ms ({overhead_ms:.2f}ms per call)")
```

Comment:
These calculations and prints should be moved after the synchronize.
Should only save the current time here to minimize the impact your code has on the measured time and do the calculations/prints after synchronize.
```
    start_time = time.time()
    for _ in range(num_call_to_stack):
        op_func()
    dispatch_end_time = time.time()
    tt_lib.device.Synchronize(device)
    duration = 1000 * (time.time() - dispatch_end_time)
    overhead = 1000 * (dispatch_end_time - start_time)
    ...
```

Reply:
Goog sugestion, ill change in next pull req.